tests/lib/prepare-project.sh

```diff
@@ -32,6 +44,28 @@ build_deb(){
     cp ../*.deb "$GOHOME"
 }
 
+fedora_build_rpm() {
```

Comment:
Yes, we can strip out a generic portion. I would like to get specific `fedora_build_rpm` and `opensuse_build_rpm` functions in for now as this is already waiting for too long. There will be a follow up PR to combine as much as possible.